kyaml/setters2/settersutil/substitutioncreator.go

```diff
@@ -62,3 +73,106 @@ func (c SubstitutionCreator) Create(openAPIPath, resourcesPath string) error {
 		Outputs: []kio.Writer{inout},
 	}.Execute()
 }
+
+// CreateSettersForSubstitution creates the setters for all the references in the substitution
+// values if they don't already exist in openAPIPath file.
+func (c SubstitutionCreator) CreateSettersForSubstitution(openAPIPath string) error {
+	b, err := ioutil.ReadFile(openAPIPath)
+	if err != nil {
+		return err
+	}
+
+	// parse the yaml file
+	y, err := yaml.Parse(string(b))
+	if err != nil {
+		return err
+	}
+
+	m, err := c.GetValuesForMarkers()
+	if err != nil {
+		return err
+	}
+
+	// for each ref in values, check if the setter already exists, if not create them
+	for _, value := range c.Values {
+		obj, err := y.Pipe(yaml.Lookup(
+			// get the setter key from ref. Ex: from #/definitions/io.k8s.cli.setters.image_setter
+			// extract io.k8s.cli.setters.image_setter
+			"openAPI", "definitions", strings.Split(value.Ref, "/")[2]))
+
+		if err != nil {
+			return err
+		}
+
+		if obj == nil {
+			sd := setters2.SetterDefinition{
+				// get the setter name from ref. Ex: from #/definitions/io.k8s.cli.setters.image_setter
+				// extract image_setter
+				Name:        strings.Split(value.Ref, ".")[4],
```

Comment:
Why not use TrimPrefix?

Reply:
Looks like it still isn't using TrimPrefix?

Reply:
Sorry. Missed it. Done now.